### PR TITLE
libjpeg: use GitHub for checkout and updates

### PR DESCRIPTION
**Pull request by luhring**
@rawlingsj per our convo earlier today! But I think I did something wrong, judging by https://github.com/wolfi-dev/os/actions/runs/4825705527/jobs/8596819923

Any help is appreciated 😄

### CODE REVIEW

File: libjpeg-turbo.yaml [hunk 1]
```diff
@@ -60,5 +61,6 @@ subpackages:
 
 update:
   enabled: true
-  release-monitor:
-    identifier: 1648
+  github:
+    identifier: libjpeg-turbo/libjpeg-turbo
+    use-tag: true
```
Comment (rawlingsj) [inline libjpeg-turbo.yaml:66]:
```suggestion
    use-tag: true
    tag-filter: 2.
```

**Comment by rawlingsj** [comment]
sure thing - I made a suggestion in the comment above which uses `tag-filter:`, you can then verify locally using
```
wolfictl check update  --override-version 0.0.1 libjpeg-turbo

2023/04/28 09:47:04 wolfictl check update: cloning sources from https://github.com/libjpeg-turbo/libjpeg-turbo tag 2.1.91 into a temporary directory, this may take a while
Enumerating objects: 5697, done.
Counting objects: 100% (5697/5697), done.
Compressing objects: 100% (2568/2568), done.
Total 5697 (delta 4017), reused 4375 (delta 2998), pack-reused 0
2023/04/28 09:47:08 wolfictl check update: git-checkout was successful
```

**Comment by luhring** [comment]
@rawlingsj I see! So is the guidance that if _any of the tags_ from the repo can't be parsed as versions, a `tag-filter` is needed?

**Review by rawlingsj — APPROVED** [review]

**Comment by rawlingsj** [comment]
There's a couple of options to handle different use cases, there's docs here https://github.com/chainguard-dev/melange/blob/main/docs/UPDATE.md#github

`tag-filter` is passed to the github graphql API to filter on
there's now also `ignore-regex-patterns:` which allows you to pass a regex and ignore bad version strings.

**Pull request merged**